File: 240424/비오는 날/rainy-day.cpp
#include <iostream>

using namespace std;

class Data {
public:
	string date, day, weather;

	Data(string date, string day, string weather) {
		this->date = date;
		this->day = day;
		this->weather = weather;
	}
};


int main() {
	int n;
	cin >> n;

	Data result("9999-99-99", "", "Rain");

	for (int i = 0; i < n; ++i) {
		string date, day, weather;
		cin >> date >> day >> weather;
		if (weather == "Rain")
			if (date <= result.date) {
				result.date = date;
				result.day = day;
			}
	}

	cout << result.date << " " << result.day << " " << result.weather;

	return 0;
}

File: 240509/비오는 날/rainy-day.py
n = int(input())

class Data:
    def __init__(self, date, day, weather):
        self.date = date
        self.day = day
        self.weather = weather

result = Data("9999-99-99", "", "")
for _ in range(n):
    date, day, weather = input().split()

    data = Data(date, day, weather)
    if weather == "Rain":
        if result.date > data.date:
            result = data

print(result.date, result.day, result.weather)